scripts/utils/trading_strategy_helpers.js

```diff
@@ -360,6 +361,66 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
     return buildBundledTransaction(transactions)
   }
 
+  /**
+   * Batches together a collection of transfer-related transaction information. Particularily,
+   * the resulting transaction is that of transfering all specified funds from master through its brackets
+   * followed by approval and deposit of those same tokens into BatchExchange on behalf of each bracket.
+   *
+   * @param {string} masterAddress Ethereum address of Master Gnosis Safe (Multi-Sig)
+   * @param {Transfer[]} transferList List of {@link Deposit} that are to be bundled together
+   * @param {boolean} [useWei=false] flag allowing user to specify if amounts are already in Base token units
+   * @param {boolean} [unsafe=false] does not perform balance verification
+   * @param {boolean} [debug=false] prints log statements when true
+   * @returns {Transaction} all the relevant transaction information used for submission to a Gnosis Safe Multi-Sig
+   */
+  const buildTransferDataFromList = async function (masterAddress, transferList, useWei = false, unsafe = false, debug = false) {
+    const log = debug ? (...a) => console.log(...a) : () => {}
+
+    const uniqueTokens = uniqueItems(transferList.map((t) => t.tokenAddress))
+    const tokenInfo = fetchTokenInfoForFlux(transferList)
+
+    // Will be used to make sufficient balance assertion before transfer.
+    const cumulativeAmounts = new Map(uniqueTokens.map((address) => [address, new BN(0)]))
+    const transactions = await Promise.all(
+      transferList.map(async (transfer) => {
+        const token = await tokenInfo[transfer.tokenAddress]
+        let unitAmount = new BN(transfer.amount)
```

Comment:
This will fail for realistic amounts (assuming `useWei=true`) with your current JSON format.
To see what I mean, add this to `sampleTransferFile.json`:
```
  {
    "amount": 1000000000000000000000,
    "tokenAddress": "0xc778417e063141139fce010982780140aa0cd5ab",
    "receiver": "0x0000000000000000000000000000000000000005"
  }
```
Note that the amount is just 1000 for a token with 18 decimals.
That's because there are no quotes around the amount value, so it's read as a number, and `BN` rightfully fails to parse float amounts larger than `2^52`.
Note that just converting this amount to string and passing it to the same function might not be enough. Because of float precision this number could change when made into a string, and a user who deposited the exact amount in his master Safe would get an "unfixable" insufficient balance error when running the scripts even if the balance is enough.
Not sure what's best here. Since the JSON is our user interface, I'd say the script should fail if `typeof(transfer.amount)` is `number` with some helpful error like "wrap your amount in quotes" and only accept strings.

Reply:
I'd add a "large number" test to catch this in the future.

Reply:
Good to point this out! I like the `typeof` suggestion and will go with that.

Reply:
Yep, Agree with @fedgiac. I think in general, it's better to use strings, specially for the mentioned reason, and also for the error of working with floating point instead of `BN` or `BigNumber`

Note that `BigNumber` can be used to handle decimal numbers, `BN` for big integers. Both allow to create instances of them from a `string`

Example of one issue with floats that we don't get if we use `BigNumber` for example:
![image](https://user-images.githubusercontent.com/2352112/84783212-48b0ab80-afe9-11ea-8c50-7ec37390ea42.png)

Reply:
I have decided to remove useWEI all together because it was not worth the extra complexity.